libs/src/api/Track.ts

```diff
@@ -562,6 +573,68 @@ export class Track extends Base {
     }
   }
 
+  /**
+   * Creates a trackId for each CID in metadataCids and adds each track to chain for this user.
+   */
+  async addTracksToChainV2(trackMetadatas: TrackMetadata[]) {
+    const ownerId = this.userStateManager.getCurrentUserId()
+    if (!ownerId) {
+      throw new Error('No users loaded for this wallet')
+    }
+
+    let trackIds: number[] = []
+    let requestFailed = false
+    await Promise.all(
+      trackMetadatas.map(async (trackMetadata) => {
+        try {
+          const { trackId } = await this.writeTrackToChain(
+            trackMetadata,
+            Action.CREATE
+          )
+          trackIds.push(trackId)
+        } catch (e) {
+          requestFailed = true
+          console.error(`Failed to add track to chain: ${e}`)
+        }
+      })
+    )
+
+    // Any failures in addding track to the blockchain will prevent further progress.
+    // The list of successful track uploads is returned for revert operations by caller
+    trackIds = trackIds.filter(Boolean)
+    const error: boolean =
```

Comment:
prob dont need to explicitly type this, if so just wrap everything in `Boolean`

Reply:
you right, I was just making it explicit because error is usually not a boolean elsewhere. but it's easy to see from the code and IDE so I removed it now ✅